spring-cloud-gcp-storage/src/main/java/org/springframework/cloud/gcp/storage/GoogleStorageResourceBucket.java

```diff
@@ -0,0 +1,163 @@
+/*
+ *  Copyright 2017 original author or authors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.springframework.cloud.gcp.storage;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URL;
+
+import com.google.cloud.storage.Bucket;
+import com.google.cloud.storage.BucketInfo;
+import com.google.cloud.storage.Storage;
+
+import org.springframework.core.io.Resource;
+import org.springframework.core.io.WritableResource;
+import org.springframework.util.Assert;
+
+/**
+ * Implements {@link WritableResource} for buckets in Google Cloud Storage.
+ */
+public class GoogleStorageResourceBucket implements WritableResource {
```

Comment:
WritableResource not needed here?

Reply:
I was thinking of keeping it there just so we can throw the unsupported operation exception along with the explanation string that it's a bucket resource. The reason being that the user will be using bucket path resources and object path resources in the same way, and if they inadvertently try to get an output stream or check isWriteable I would rather they get the unsupported operation exception with an explanation than a class casting exception because those two functions don't exist at all.

Reply:
Actually, the way things are now, the user needs to cast a `Resource` into a `WritableResource` to be able to write to it. In which case, the cast should actually just fail, instead of allowing it and then throwing UOE.